Add Pangea provider to Guardrails hook

ryanmeans opened the pull request:
## Title

Add Pangea provider to Guardrails hook

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

🆕 New Feature

## Changes

Adds [Pangea AI Guard](https://pangea.cloud/docs/ai-guard/apis) to guardail hooks

Supports prompt injection on both the request and response to the upstream LLM. 


## Test Screenshot
![Screenshot 2025-05-12 at 4 12 45 PM](https://github.com/user-attachments/assets/4ce53660-cc98-4e7d-9d65-835cd2b5a585)

mweinberger committed:
added Pangea as a guardrail vendor

ryanmeans committed:
Adding output recipe, cleaning up some imports

ryanmeans committed:
Add Pangea guardrails tests

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/BerriAI/litellm?pullRequest=10775) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/BerriAI/litellm?pullRequest=10775) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: Ryan Means<br/>:x: mweinberger<br/><hr/>**Ryan Means** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/BerriAI/litellm?pullRequest=10775) it.</sub>

krrishdholakia commented:
Hi @ryanmeans thanks for this PR. Just confirming - were you able to verify this worked for you via manual qa as well?

krrishdholakia commented:
You should also probably add docs for this in here - https://github.com/BerriAI/litellm/tree/5fbcd2260499d525d0e3f621cdc6dc8816412452/docs/my-website/docs/proxy/guardrails

And in the sidebar - https://github.com/BerriAI/litellm/blob/5fbcd2260499d525d0e3f621cdc6dc8816412452/docs/my-website/sidebars.js#L149

mweinberger committed:
Add docs and sidebar

krrishdholakia commented:
@ryanmeans can you fix merge conflicts so we can merge this in?

ryanmeans committed:
Move to use async_precall_hook instead of moderation hook

ryanmeans commented:
Hey @krrishdholakia, thanks for keeping track of this. We are going with the recommendation to use the `async_pre_call_hook` instead of the `async_moderation_hook`, so I'm converting to draft. It should be ready pretty soon though. 

And to confirm, yes, I've verified this with manual QA against a local litellm instance as well.

ryanmeans commented:
@krrishdholakia as a follow up question, something I've noticed is that the hooks don't normalize the request / response content. 

That is to say, the hook will run on `/v1/chat/completions`, `/v1/completions`, and `/v1/responses`, `/v1/messages`, etc, but these each have different response and request formats. What is the expectation from the guard rails here? Looking at other plugins, it doesn't seem like there is a great solution, other than just handle / parse them specifically. It seems like this is a detail that the guard rail shouldn't need to be concerned with.

Even in the /v1/chat/completions endpoint, You can call w/ something like:
```
{"model": "gpt-3.5-turbo", "messages": [{"role": "user", "content": [{"type": "text", "text": "Hello!"}]}]}
```
At least glancing at their source code, it seems like existing guardrail's would fail / skip on this kind of call.

ryanmeans committed:
Update to "new" format (accept mode)

Add response for redaction, support transforming request / response
based off the original type of call (/v1/completions,
/v1/chat/completions)

ryanmeans committed:
Fix tests

ryanmeans committed:
Fix unused imports

krrishdholakia commented:
Looks like your doc has invalid markdown @ryanmeans 

<img width="1391" alt="Screenshot 2025-05-20 at 5 49 21 PM" src="https://github.com/user-attachments/assets/d62c67a6-7ca3-4b60-8bd1-79906dc10bae" />

ryanmeans committed:
Fix .md

ryanmeans commented:
Thanks, I wasn't able to look at the Vercel check to see why it was failing .

ryanmeans commented:
@krrishdholakia Any updates here? Are you looking for anything else from our MR?
